crates/core/tedge/src/cli/connect/c8y_direct_connection.rs

```diff
@@ -111,22 +112,34 @@ fn publish_device_create_message(
 
 fn load_root_certs(
     root_store: &mut rustls_0_19::RootCertStore,
-    cert_dir: FilePath,
+    cert_path: FilePath,
 ) -> Result<(), ConnectError> {
-    for file_entry in fs::read_dir(cert_dir)? {
-        let file = file_entry?;
-        let f = File::open(file.path())?;
-        let mut rd = BufReader::new(f);
-        let _ = root_store.add_pem_file(&mut rd).map(|_| ()).map_err(|()| {
-            Error::new(
-                ErrorKind::InvalidData,
-                "could not load PEM file".to_string(),
-            )
-        });
+    if fs::metadata(&cert_path)?.is_dir() {
+        for file_entry in fs::read_dir(cert_path)? {
+            let file = file_entry?;
+            add_root_cert(root_store, file.path())?;
+        }
+    } else {
+        add_root_cert(root_store, cert_path.into())?;
     }
     Ok(())
 }
 
+fn add_root_cert(
+    root_store: &mut rustls_0_19::RootCertStore,
+    cert_path: PathBuf,
+) -> Result<(), ConnectError> {
+    let f = File::open(cert_path)?;
+    let mut rd = BufReader::new(f);
+    let _ = root_store.add_pem_file(&mut rd).map(|_| ()).map_err(|()| {
```

Comment:
This `rusttls` library doesn't have an API that supports a directory of certificates, is it? That's really unfortunate.

Reply:
By default it reads the `/etc/ssl/certs` but it can not read `self-signed` cert that's why it has to be added explicitly to the `rustls` root_cert_store.

Reply:
What is really unfortunate is that this [`add_pem_file`](https://docs.rs/rustls/0.19.1/rustls/struct.RootCertStore.html#method.add_pem_file) method no more exists in the new versions of `rustls`. We must find another way to load the certificates - but not in this PR.

Reply:
Agree, maybe we can use https://docs.rs/rustls/latest/rustls/struct.RootCertStore.html#method.add method to add a certificate to the root_store